### ISSUE TITLE
Request / suggestion / angular

### ISSUE THREAD

**Issue by jbdioli** [mattiasw/ExifReader#69]
Hello

Can you make it available for angular 7 and above ?

thanks

**Comment by mattiasw** [comment]
Hi! I've not used Angular in many years so I'm not sure what this means. Can you explain how you would want to use ExifReader and what is not working with that setup?

**Comment by jbdioli** [comment]
I will try to explain correctly, I'm new in programming.

If I want to "use it"  I have to do the import like this:

`import * as ExifReader from 'exifreader';`

 instead of like this, with a constructor:

`
import { ExifReader } from '@exifreader'

constructor(private exifReader: ExifReader) {}
`

and I have difficulty to found out how to convert what I'm receiving from the template, that will be an type element File to an ArrayBuffer.
I think you fileBuffer is an ArrayBuffer .... it's not very clear in you documentation.

It will be wonderful if we can give directly the element File.

**Comment by mattiasw** [comment]
I think the ES module example would be helpful in your case: https://github.com/mattiasw/ExifReader/tree/master/examples/esm

I sounds like the line you're looking for is `reader.readAsArrayBuffer(files[0]);` where you probably would replace `files[0]` with the File you get from the template.

The reason ExifReader doesn't use file objects directly is because this is supposed to be a cross-platform library that doesn't care if it's run in a browser, node, or gets the file contents from somewhere else. So it's a bit out-of-scope for that. But the examples should help getting it set up.

I think the import line should be like this even in your Angular setup: `import ExifReader from 'exifreader';`

I believe you're correct about the documentation and `fileBuffer`. I will try to add some more details to the README to make it more clear, and refer to the examples.

**Comment by jbdioli** [comment]
Yes files[0] as to be converted as you said and show in your example. I did it a little bit different but it's exactly in this way... (I will show you when I will be done with a "nice" code :D )

For the import it doesn't work. It shows this error:

> import ExifReader
> Module '"c:/Project/IONIC/Exif/node_modules/exifreader/exif-reader"' has no default export.ts(1192)

**Comment by mattiasw** [comment]
Not sure why it gives that error. It works fine for me. My guess is that it has something to do with TypeScript/your bundler doing something different than mine. Use what works. :-)

**Comment by KrakenZ** [comment]
I have the same issue. Has anyone found a solution to this?

**Comment by mattiasw** [comment]
What exactly is that issue? In the discussion above a solution was found and it was just a matter of importing the right way and using the file data correctly (which may differ between Angular/TypeScript and vanilla JS, not sure since I don't use them).

**Comment by KrakenZ** [comment]
The problem is fixed, but I'm not sure why:
`import ExifReader from 'exifreader';` does not work:
it says `Module '".../node_modules/exifreader/exif-reader"' has no default export.` (I replaced the full path with "...")

With `import * as ExifReader from 'exifreader';`, it first did not work. Same error. Even after I restarted the Angular compilation, even after checking the node modules. I re-installed it a few times and now it seems to work.
